### PR TITLE
Report timing on failed executions too

### PR DESCRIPTION
**Pull request by xster**
#9499

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -153,31 +151,38 @@ abstract class FlutterCommand extends Command<Null> {
...
-        // the duration of the entire execution.
-        (commandResult?.endTimeOverride ?? endTime).difference(startTime), 
-        // Report in the form of `success-[parameter1-parameter2]`, all of which
-        // can be null if the command doesn't provide a FlutterCommandResult.
-        label: label == '' ? null : label,
-      );
+    FlutterCommandResult commandResult;
+    try {
+      commandResult = await verifyThenRunCommand();
+    } on ToolExit catch (_) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: you can remove the `catch (_)` part

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 2]
```diff
@@ -153,31 +151,38 @@ abstract class FlutterCommand extends Command<Null> {
...
+      commandResult = const FlutterCommandResult(ExitStatus.fail);
+      rethrow;
+    } finally {
+      final DateTime endTime = clock.now();
+      printTrace("'flutter $name' took ${getElapsedAsMilliseconds(endTime.difference(startTime))}.");
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: flip the `"` and `'` over so the output uses `"` and the string itself uses `'`.

File: packages/flutter_tools/test/src/runner/flutter_command_test.dart [hunk 3]
```diff
@@ -113,20 +115,47 @@ void main() {
...
+      // Crash if called a third time which is unexpected.
+      mockTimes = <int>[1000, 2000];
+
+      final DummyFlutterCommand flutterCommand = 
+          new DummyFlutterCommand(commandFunction: () async { throwToolExit('fail'); });
+
+      try {
+        await flutterCommand.run();
+        fail('Mock should make this fail');
+      } on ToolExit catch (_) {
```
Comment (Hixie) [inline packages/flutter_tools/test/src/runner/flutter_command_test.dart]: ditto

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (xster) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: oh very cool, thanks

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 2 above)
Comment (xster) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: ok

File: packages/flutter_tools/test/src/runner/flutter_command_test.dart (hunk 3 above)
Comment (xster) [inline packages/flutter_tools/test/src/runner/flutter_command_test.dart]: done

**Comment by Hixie** [comment]
LGTM.

fyi @tvolkert

**Comment by tvolkert** [comment]
LGTM

**Pull request merged**